src/components/Lottery/index.js

```diff
@@ -0,0 +1,127 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Field, Form } from 'react-final-form';
+import AuthContext from '../../context/Authentication';
+import { useUpdateUser } from '../../hooks/Api/Users/Update';
+import { InlineButton } from '../Forms/Button';
+import DonationForm from '../Forms/DonationForm';
+import { FinallyMessage } from '../Forms/FinallyMessage';
+import { TextInputWrapped } from '../Forms/TextInput';
+import { CTAButton, CTAButtonContainer } from '../Layout/CTAButton';
+import { RequestLoginCodeWithEmail } from '../Login/RequestLoginCode';
+import * as s from './style.module.less';
+
+export const Lottery = () => {
+  const [wantsToDonate, setWantsToDonate] = useState(null);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const { isAuthenticated, customUserData: userData } = useContext(AuthContext);
+  const [updateUserState, updateUser] = useUpdateUser();
+
+  useEffect(() => {
+    // If user has signed in and does not have username, we want to
+    // sign em up for the lotto right away.
+    if (
+      isAuthenticated &&
+      userData.username &&
+      wantsToDonate !== null &&
+      !hasSubmitted
+    ) {
+      updateUser({ lottery: '2021' });
+      // We need to set a flag, otherwise it will trigger multiple times,
+      // while user goes through onboarding
+      setHasSubmitted(true);
+    }
+  }, [isAuthenticated, wantsToDonate, userData]);
+
+  if (wantsToDonate === null) {
+    return (
+      <div className={s.container}>
+        <CTAButtonContainer>
+          <CTAButton
+            onClick={() => {
+              setWantsToDonate(true);
+            }}
+            size="MEDIUM"
+          >
+            Spenden und an der Verlosung teilnehmen
+          </CTAButton>
+        </CTAButtonContainer>
+        <p className={s.buttonText}>
+          <InlineButton
+            onClick={() => {
+              setWantsToDonate(false);
+            }}
+          >
+            Nur an der Verlosung teilnehmen
+          </InlineButton>
+        </p>
+      </div>
+    );
+  }
+
+  if (wantsToDonate !== null && !isAuthenticated) {
+    return (
+      <RequestLoginCodeWithEmail>
+        <h3 className={s.headingWhite}>
+          Hey! Schön, dass du da bist. Hier geht's zum Login.
+        </h3>
+      </RequestLoginCodeWithEmail>
+    );
+  }
+
+  if (updateUserState === 'loading') {
+    return (
+      <FinallyMessage state="progress">
+        Los wird generiert, bitte einen Moment Geduld...
```

Comment:
Is it expected, that the "lottery ticket" is sent before the user has donated?

Reply:
yes, because, if the user wants to donate via paypal or transaction we don't know, when they are done.

Reply:
I'll edit the wording.